src/main/java/com/amazon/opendistroforelasticsearch/sql/domain/Having.java

```diff
@@ -87,12 +98,16 @@ public void explain(AggregationBuilder groupByAgg, List<Field> fields) throws Sq
             return;
         }
 
+        // parsing the fields from SELECT and HAVING clause
+        List<Field> combinedFields = Stream.of(fields, getHavingFields())
```

Comment:
not necessary, if you like, we can avoid creating a new list by Guava's `Iterables.concat(list1, list2)` and pass result as Iterable view to `contextForFieldsInSelect`.

Reply:
Done.